### PR TITLE
Make SnapshotsInProgress.Entry#indices a Map

### PR DESCRIPTION
**Pull request by original-brownbear**
We constantly need to look up from index name to `IndexId`, might as well
just have this field as a map to simplify some code here and there and save
a few cycles during cluster state updates for snapshots of large index counts.

### CODE REVIEW

**Review by fcofdez — APPROVED** [review]
LGTM 👍

File: server/src/main/java/org/elasticsearch/repositories/RepositoryData.java [hunk 1]
```diff
@@ -555,10 +555,11 @@ public IndexId resolveIndexId(final String indexName) {
...
-    public List<IndexId> resolveIndices(final List<String> indices) {
-        List<IndexId> resolvedIndices = new ArrayList<>(indices.size());
+    public Map<String, IndexId> resolveIndices(final List<String> indices) {
+        Map<String, IndexId> resolvedIndices = new HashMap<>(indices.size());
         for (final String indexName : indices) {
-            resolvedIndices.add(resolveIndexId(indexName));
+            final IndexId indexId = resolveIndexId(indexName);
+            resolvedIndices.put(indexId.getName(), indexId);
         }
         return resolvedIndices;
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/RepositoryData.java:564]: Can be unmodifiable map I think

File: server/src/main/java/org/elasticsearch/repositories/RepositoryData.java [hunk 2]
```diff
@@ -580,7 +581,7 @@ public IndexId resolveIndexId(final String indexName) {
             if (indexId == null) {
                 indexId = new IndexId(index, UUIDs.randomBase64UUID());
             }
-            snapshotIndices.add(indexId);
+            snapshotIndices.put(indexId.getName(), indexId);
         }
         return snapshotIndices;
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/RepositoryData.java:586]: Can be unmodifiable map I think

**Review by tlrx — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java [hunk 3]
```diff
@@ -561,7 +563,18 @@ private Entry(StreamInput in) throws IOException {
...
             partial = in.readBoolean();
             state = State.fromValue(in.readByte());
-            indices = in.readList(IndexId::new);
+            final int indexCount = in.readVInt();
+            if (indexCount == 0) {
+                indices = Collections.emptyMap();
+            } else {
+                final Map<String, IndexId> idx = new HashMap<>(indexCount);
+                for (int i = 0; i < indexCount; i++) {
+                    final String name = in.readString();
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:572]: Can we use `new IndexId(in)` instead, to ensure any wire protocol changes to `IndexId` are centralized?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java (hunk 3 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:572]: ++

File: server/src/main/java/org/elasticsearch/repositories/RepositoryData.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/repositories/RepositoryData.java:564]: ++

File: server/src/main/java/org/elasticsearch/repositories/RepositoryData.java (hunk 2 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/repositories/RepositoryData.java:586]: ++

**Comment by original-brownbear** [comment]
Thanks everyone!

**Pull request merged**